ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionValidator.java

```diff
@@ -109,6 +116,11 @@ public MainnetTransactionValidator(
       return signatureResult;
     }
 
+    if (chainId.isPresent() && transaction.getChainId().isEmpty() && requireTxReplayProtection) {
```

Comment:
Thought about this some more - this is actually the wrong place to validate.  This validator is executed on transactions received over gossip, and during block processing.  If we validate here, we're going to be unable to process blocks that contain unprotected transactions, and we're going to censor gossiped transactions.  We should move this validation to the JSON-RPC layer.

Reply:
Ah, it was initially in the `TransactionPool` class under the `addLocalTransaction` method.

Reply:
Would be good to get test coverage showing we can process blocks with a mix of un/protected txs with this option enabled.